### PR TITLE
[20091] Add actions to support Fast DDS CI on macOS

### PR DESCRIPTION
**Pull request by EduPonz**
This PR adds/modifies the necessary actions to support running Fast DDS CI on macOS Github-hosted runners.

* Related to eprosima/fast-dds#4100
* Related to eprosima/fast-dds#4109
* Related to eprosima/fast-cdr#197

### CODE REVIEW

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
I suggest updating the README with the new feature.

File: multiplatform/colcon_test/action.yml [hunk 1]
```diff
@@ -29,8 +29,7 @@ inputs:
 
   workspace:
     description: 'Workspace where built has been done'
-    required: false
-    default: ''
+    required: true
```
Comment (JesusPoderoso) [inline multiplatform/colcon_test/action.yml:32]: This may break some other usages of eProsima-CI (in other repos different than Fast-DDS)

**Review by JesusPoderoso — APPROVED** [review]
LGTM

File: multiplatform/colcon_test/action.yml (hunk 1 above)
Comment (JesusPoderoso) [reply multiplatform/colcon_test/action.yml:32]: Internally agreed on set the default value as the platform-specific action ([ubuntu](https://github.com/eProsima/eProsima-CI/blob/5bcd96869682934fbde5239b55bfd44362c57e93/ubuntu/colcon_test/action.yml#L33C14-L33C39), [windows](https://github.com/eProsima/eProsima-CI/blob/5bcd96869682934fbde5239b55bfd44362c57e93/windows/colcon_test/action.yml#L33C14-L33C39))
```suggestion
    '${{ github.workspace }}'
    required: false
```

**Review by JLBuenoLopez-eProsima — COMMENTED** [review]
Thanks for updating the README

**Review by JesusPoderoso — APPROVED** [review]
LGTM
Please, remember to update all `@feature/fastdds_mac_ci_support` to `@main` before merging

**Review by JesusPoderoso — APPROVED** [review]

**Pull request merged**